nurseconnect/views.py

```diff
@@ -26,8 +41,55 @@ def search(request, results_per_page=10):
     except EmptyPage:
         search_results = paginator.page(paginator.num_pages)
 
-    return render(request, "search/search_results.html", {
+    return render(request, "search/search.html", {
+        "searched": True,
         "search_query": search_query,
         "search_results": search_results,
         "results": results,
     })
+
+
+class RegistrationView(FormView):
+    form_class = forms.RegistrationForm
+    template_name = "profiles/register.html"
+
+    def form_valid(self, form):
+        username = form.cleaned_data["username"]
+        password = form.cleaned_data["password"]
+        user = User.objects.create_user(username=username, password=password)
+        user.save()
+
+        authed_user = authenticate(username=username, password=password)
+        login(self.request, authed_user)
+        return HttpResponseRedirect(form.cleaned_data.get("next", "/"))
```

Comment:
This doesn't redirect to the correct URL.
it loads the correct template, yes, but can't we rather go straight to the homepage URL  instead?

You can do specify `success_url` under `template_name` and then instead of `return HttpResponseRedirect ...` do 
`return super(RegistrationView, self).form_valid(form)`. Or stick to the `HttpResponseRedirect` but use the home page url. :)